### ISSUE TITLE
Possible to pick/omit?

### ISSUE THREAD

**Issue by Kolby-Udacity** [saiichihashimoto/sanity-typed-schema-builder#210]
Hi, I'm trying to utilize my existing sanity-typed-schema-builder document to create a new Zod document. I want to pick fields from the former to add be added to the latter. Is this possible somehow?

Example:

```
export const programSchema = s.document({
  name: 'program',
  title: 'Program',
  fields: [
    {
      name: 'title',
      title: 'Title',
      description: 'The title of the program',
      type: s.string(),
    },
    {
      name: 'description',
      title: 'Description',
      description: 'The description of the program',
      type: s.string(),
    }
  ],
});

export const programPreview = z.object({
  title: programSchema.pick({title: true})
});
```

Thanks

**Comment by saiichihashimoto** [comment]
```typescript
const zodOfDocument: ZodType = s.document({ ... }).zod;

// Do what you want with zodOfDocument
```

**Comment by Kolby-Udacity** [comment]
> ```ts
> const zodOfDocument: ZodType = s.document({ ... }).zod;
> 
> // Do what you want with zodOfDocument
> ```

Thanks, I tried that, but TypeScript says:

`Property 'pick' does not exist on type ZodType ...`

**Comment by saiichihashimoto** [comment]
can you show me a repro? Your code snippet has `programSanitySchema.pick({title: true})` which is incorrect.

**Comment by Kolby-Udacity** [comment]
Here is a codesandbox: https://codesandbox.io/s/silent-thunder-7pg960?file=/src/App.tsx

<img width="1161" alt="image" src="https://user-images.githubusercontent.com/92734750/231265409-faca3776-8e8a-428b-bae2-fcbbe3179373.png">

**Comment by saiichihashimoto** [comment]
ah, I see. the object returned is of type `ZodType`, not `ZodObject`, so it doesn't have the object specific methods. I think that's a change we can make!

**Comment by saiichihashimoto** [comment]
FYI until we fix it, you should just be able to cast it into a ZodObject. I think it actually is a ZodObject so this shouldn’t be an issue.

**Comment by Kolby-Udacity** [comment]
Yeah I actually tried that for awhile the other day. I settled on using `as ZodObject<any>`, which works, but doesn't give me the options when I use `pick`. Not a huge problem while I wait for a fix, but I'm curious if there is a way to fix this. Seems the only way (other than recreating a Zod object) is `as ZodObject<programSchema.zod>`, but that obviously doesnt work because if it did none of this would be a problem lol

**Comment by schietegal** [comment]
Tried a whole bunch but i don't get it to work.

Same problem if you are trying to follow references the given .parse method is unusable
e.g. :
```typescript
const data = await sanity_client.fetch(`//groq
    *[_type == "jobs"]{
        ...,
        kategorie[]->,
        modell[]->,
        ansprechpartner->
    }`);

const parsed_data = jobs.zod.array().parse(data); //Error 'invalid_literal' bc of

```
